### PR TITLE
update cron integration test

### PR DESCRIPTION
**Pull request by yux0**
update cron integration test

### CODE REVIEW

File: host/integration_test.go [hunk 1]
```diff
@@ -1017,6 +1010,15 @@ func (s *integrationSuite) TestCronWorkflow() {
 	s.Nil(err)
 	expectedExecutionTime := dweResponse.WorkflowExecutionInfo.GetStartTime() + 3*time.Second.Nanoseconds()
 	s.Equal(expectedExecutionTime, dweResponse.WorkflowExecutionInfo.GetExecutionTime())
+
+	// The first execution is the termination execution. It verified below.
+	firstExecutionTime := time.Unix(0, dweResponse.WorkflowExecutionInfo.GetStartTime())
```
Comment (yycptt) [inline host/integration_test.go]: firstStartTime.

Comment (yux0) [reply host/integration_test.go]: updated

File: host/integration_test.go [hunk 2]
```diff
@@ -27,7 +27,9 @@ import (
 	"errors"
 	"flag"
 	"fmt"
+	"github.com/uber/cadence/common/backoff"
```
Comment (yycptt) [inline host/integration_test.go]: Move this to the second import section?

File: host/integration_test.go [hunk 3]
```diff
@@ -1017,6 +1010,19 @@ func (s *integrationSuite) TestCronWorkflow() {
 	s.Nil(err)
 	expectedExecutionTime := dweResponse.WorkflowExecutionInfo.GetStartTime() + 3*time.Second.Nanoseconds()
 	s.Equal(expectedExecutionTime, dweResponse.WorkflowExecutionInfo.GetExecutionTime())
+
+	sort.Slice(closedExecutions, func(i, j int) bool {
+		return closedExecutions[i].GetStartTime() < closedExecutions[j].GetStartTime()
+	})
+	// The first execution is the termination execution.
```
Comment (yycptt) [inline host/integration_test.go]: Does this comment still apply?

File: host/integration_test.go [hunk 4]
```diff
@@ -1968,18 +1971,25 @@ func (s *integrationSuite) TestCronChildWorkflowExecution() {
 		time.Sleep(200 * time.Millisecond)
 	}
 	s.NotNil(closedExecutions)
-	firstExecutionTime := closedExecutions[0].GetExecutionTime()
-	for i := 1; i != 4; i++ {
+
+	sort.Slice(closedExecutions, func(i, j int) bool {
+		return closedExecutions[i].GetStartTime() < closedExecutions[j].GetStartTime()
+	})
+	// The first execution is the termination execution.
```
Comment (yycptt) [inline host/integration_test.go]: and this one?

File: host/integration_test.go [hunk 5]
```diff
@@ -1017,6 +1010,19 @@ func (s *integrationSuite) TestCronWorkflow() {
...
+	lastExecution := closedExecutions[0]
+	for i := 1; i != 4; i++ {
+		executionInfo := closedExecutions[i]
+		backoffDuration := backoff.GetBackoffForNextSchedule(cronSchedule, time.Unix(0, lastExecution.GetStartTime()), time.Unix(0, lastExecution.GetCloseTime()))
+		// Round up to compare only seconds
+		s.Equal(backoffDuration.Nanoseconds()/1000000000, int64(executionInfo.GetExecutionTime()-lastExecution.GetCloseTime())/1000000000)
```
Comment (yycptt) [inline host/integration_test.go]: `backoffDuration` is basically the `firstDecisionBackoffTime` and when doing describe workflow, the execution time is calculated as `startTime` + `firstDecisionBackoffTime`. So why we use `closeTime` here?

Comment (yux0) [reply host/integration_test.go]: When the instance is slow, the close time and start time could be different and we actually use closeTime to calculate the backoff duration

File: host/integration_test.go (hunk 4 above)
Comment (yux0) [reply host/integration_test.go]: Removed this comment

File: host/integration_test.go (hunk 3 above)
Comment (yux0) [reply host/integration_test.go]: Yes.

File: host/integration_test.go (hunk 2 above)
Comment (yux0) [reply host/integration_test.go]: updated

File: host/integration_test.go (hunk 3 above)
Comment (yux0) [reply host/integration_test.go]: Removed the comment

File: host/integration_test.go (hunk 5 above)
Comment (yux0) [reply host/integration_test.go]: Synced offline. The validation should be current-execution execution time - current-execution start time

**Pull request merged**